Fix Chrome AudioContext issue

amatuni opened the pull request:
New versions of Chrome prevent automatic audio playback. This implements the same fix as in issue #518

jodeleeuw commented:
Thanks @amatuni! The fix in #518 is in the main jspsych.js file to try and generalize the solution. Was that not working with this plugin? I'm trying to figure out if we need to merge this here or if I need to revisit the general solution to address it.

amatuni commented:
It's a strange situation. The current master version of jspsych was autoplaying audio in latest Chrome, and then for some reason stopped working. This happened after I introduced trial_duration logic (taken from existing plugin examples) in a custom plugin I'm using. It may be that some aspect of `end_trial()` or the trial_duration timeout (which I added to custom plugin) is somehow interacting with the audio-button-response plugin. I don't know why this would be the case, as the custom plugin and the audio-button-response should be insulated from one another? I saw that the fix from #518 was in the main jspsych.js file, and wasn't sure why this wasn't working. When I added the `context.resume()` in the audio-button-response plugin it started working again.

amatuni commented:
should also mention, this is not an issue in Firefox. Whatever the custom plugin is doing, it's not interacting with the audio-button-response plugin in Firefox
